rest_framework_json_api/filters/__init__.py

```diff
@@ -0,0 +1,6 @@
+import pkgutil
+from .sort import JSONAPIOrderingFilter  # noqa: F401
+# If django-filter is not installed, no-op.
+if pkgutil.find_loader('django_filters') is not None:
+    from .django_filter import DjangoFilterBackend  # noqa: F401
+del pkgutil
```

Comment:
Do you know any popular Django/DRF library which supports 3rd party libraries this way?

All the ones I know create a separate module/folder for 3rd party dependencies. This actually includes Django Filters (separate folder for rest_framework) or also [graphene-django](https://github.com/graphql-python/graphene-django).

I would really prefer it when we also create its own module for 3rdParty application where necessary so in this case `django_filters`.

Reply:
OK, I can move it.

Reply:
Interesting, that's where you found this code bit...

When looking at the discussion why this was introduced is that following code will work:
```python
import django_filters
django_filters.rest_framework.DjangoFilterBackend()
```

Instead if this import mechanism were not there following would need to be done to use it:
```python
import django_filters.rest_framework
django_filters.rest_framework.DjangoFilterBackend()
# or
from django_filters import rest_framework
rest_framework.DjangoFilterBackend
```

This is an API design decision by django filters that one import statement is enough to auto import all modules. We at this point do not have such an auto import mechanism in DJA and I would prefer to leave it this way.

Overall Django Filters still has `rest_framework` specific classes in its own module. As outlined above we should do the same as this way it is clear by the code that in this directory a dependency is needed to be installed.

And as we do not have the same API decision as django filters we also do not need `pkgutil`.

Reply:
@sliverc Take a look and see if this is OK. I made a `rest_framework_json_api.django_filters` next to `rest_framework_json_api.filters`. The former contains `DjangoFilterBackend` in `backends.py` and the latter current contains `OrderingFilter` but will add `QueryValidationFilter` (based off standard DRF filters) in a future PR.